bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -0,0 +1,749 @@
+static std::optional<Eigen::Isometry3d> ResolveSdfPose(
+  const ::sdf::SemanticPose &_semPose)
+{
+  math::Pose3d pose;
+  ::sdf::Errors errors = _semPose.Resolve(pose);
+  if (!errors.empty())
+  {
+    if (!_semPose.RelativeTo().empty())
+    {
+      gzerr << "There was an error in SemanticPose::Resolve:\n";
+      for (const auto &err : errors)
+      {
+        gzerr << err.Message() << std::endl;
+      }
+    }
+
+    return std::nullopt;
+  }
+
+  return math::eigen3::convert(pose);
+}
+
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfWorld(
+    const Identity &_engine,
+    const ::sdf::World &_sdfWorld)
+{
+  const Identity worldID = this->ConstructEmptyWorld(_engine, _sdfWorld.Name());
+
+  const WorldInfoPtr &worldInfo = this->worlds.at(worldID);
+
+  auto gravity = _sdfWorld.Gravity();
+  worldInfo->world->setGravity(btVector3(gravity[0], gravity[1], gravity[2]));
+
+  for (std::size_t i = 0; i < _sdfWorld.ModelCount(); ++i)
+  {
+    const ::sdf::Model *model = _sdfWorld.ModelByIndex(i);
+
+    if (!model)
+      continue;
+
+    this->ConstructSdfModel(worldID, *model);
+  }
+
+  return worldID;
+}
+
+/////////////////////////////////////////////////
+struct ParentInfo
+{
+  const ::sdf::Joint *joint;
+  const ::sdf::Model *model;
+};
+
+/////////////////////////////////////////////////
+struct Structure
+{
+  /// The root link of the model
+  const ::sdf::Link *rootLink;
+  const ::sdf::Joint *rootJoint;
+  double mass;
+  btVector3 inertia;
+
+  /// Is the root link fixed
+  bool fixedBase;
+
+  /// Get the parent joint of the link
+  std::unordered_map<const ::sdf::Link*, ParentInfo> parentOf;
+
+  /// This contains all the links except the root link
+  std::vector<const ::sdf::Link*> flatLinks;
+};
+
+/////////////////////////////////////////////////
+std::optional<Structure> ValidateModel(const ::sdf::Model &_sdfModel)
+{
+  std::unordered_map<const ::sdf::Link*, ParentInfo> parentOf;
+  const ::sdf::Link *rootLink = nullptr;
+  const ::sdf::Joint *rootJoint = nullptr;
+  bool fixed = false;
+  const std::string &rootModelName = _sdfModel.Name();
+
+  const auto checkModel =
+      [&rootLink, &rootJoint, &fixed, &parentOf, &rootModelName](
+      const ::sdf::Model &model) -> bool
+    {
+      for (std::size_t i = 0; i < model.JointCount(); ++i)
+      {
+        const auto *joint = model.JointByIndex(i);
+        const auto &parentLinkName = joint->ParentName();
+        const auto *parent = model.LinkByName(parentLinkName);
+        const auto &childLinkName = joint->ChildName();
+        const auto *child = model.LinkByName(childLinkName);
+
+        switch (joint->Type())
+        {
+          case ::sdf::JointType::FIXED:
+          case ::sdf::JointType::REVOLUTE:
+          case ::sdf::JointType::PRISMATIC:
+          case ::sdf::JointType::BALL:
+            break;
+          default:
+            gzerr << "Joint type [" << (std::size_t)(joint->Type())
```

Comment:
@mxgrey @mjcarroll What's about tell here that the joint type is not supported but instead we will create a fixed joint? This idea will remove the segfault, and the model will show up too

Reply:
by thew way a nice world to test joint is this one in `gz-sim/test/worlds/demo_joint_types.sdf`

Reply:
https://github.com/gazebosim/gz-physics/pull/404